lightbend/cloudflow
# Cloudflow flink plugin

### What changes were proposed in this pull request?


* One cloudflow application will have multiple images in future - one per runtime per project. Hence we need to have separate plugins that will build the project specific images. This PR is an implementation of `CloudflowFlinkPlugin` which will be used to develop a Flink based project in a cloudflow application.

### Why are the changes needed?

* To support runtime specific image generation of a project in a cloudflow application

### Does this PR introduce any user-facing change?

* Yes, it builds separate images for runtimes (this one is only for Flink) instead of an uber image for the whole application.

### How was this patch tested?


* Tested as follows:
  * build a base image for Flink
  * use the plugin in an example application Flink project
  * generate the image using `sbt build` for that project
  * check if the image has the streamlet descriptors as a label
